### PR TITLE
support aws websocket api gateway

### PR DESCRIPTION
**Pull request by idaunis**

### CODE REVIEW

File: src/y-websocket.js [hunk 1]
```diff
@@ -178,6 +204,7 @@ const setupWS = (provider) => {
       )
     }
     websocket.onopen = () => {
+      console.log("ooopeeenn");
```
Comment (mattstrom) [inline src/y-websocket.js]:
```suggestion
```

File: src/y-websocket.js [hunk 2]
```diff
@@ -123,23 +123,49 @@ const readMessage = (provider, buf, emitSynced) => {
...
+  /*
+  The result looks like
+  "data:application/octet-stream;base64,<your base64 data>",
+  so we split off the beginning:
+  */
+  return base64url.substring(base64url.indexOf(',')+1)
+}
+
+const toText = (data) => {
+  return new TextDecoder("latin1").decode(data)
```
Comment (mattstrom) [inline src/y-websocket.js:143]: Not UTF-8?
```suggestion
  return new TextDecoder("utf8").decode(data);
```

File: src/y-websocket.js [hunk 3]
```diff
@@ -123,23 +123,49 @@ const readMessage = (provider, buf, emitSynced) => {
...
+        data.forEach((_, i) => {
+          data[i] = event.data.charCodeAt(i);
+        });
+      }
       provider.wsLastMessageReceived = time.getUnixTime()
-      const encoder = readMessage(provider, new Uint8Array(event.data), true)
+      const encoder = readMessage(provider, data, true)
       if (encoding.length(encoder) > 1) {
-        websocket.send(encoding.toUint8Array(encoder))
+        websocket.send(toText(encoding.toUint8Array(encoder)))
```
Comment (mattstrom) [inline src/y-websocket.js:168]: Why does this line send data as `latin1`-encoded whereas the other usages of `websocket.send()` send data as `base64`?

Comment (idaunis) [reply src/y-websocket.js:168]: sorry, this is from an early attempt, we shouldn't do that

**Comment by mattstrom** [comment]
Closing in favor of #2

**Pull request closed without merging**